File: packages/kt-client/src/main/kotlin/com/google/firebase/auth/GoogleAuthProvider.kt
package com.google.firebase.auth

object GoogleAuthProvider {
    const val PROVIDER_ID = "google.com"

    @JvmStatic
    fun getCredential(idToken: String?, accessToken: String?): AuthCredential {
        return OAuthCredential(
            provider = PROVIDER_ID,
            idToken = idToken,
            accessToken = accessToken,
            rawNonce = null
        )
    }
}

File: packages/swift-client/Sources/FirebaseAuth/GoogleAuthProvider.swift
import Foundation

public enum GoogleAuthProvider {
    public static let providerID = "google.com"

    public static func credential(withIDToken idToken: String, accessToken: String) -> AuthCredential {
        OAuthCredential(providerID: providerID, idToken: idToken, rawNonce: nil, accessToken: accessToken)
    }
}
